serde/src/java/org/apache/hadoop/hive/serde2/objectinspector/primitive/PrimitiveObjectInspectorUtils.java

```diff
@@ -1124,33 +1125,25 @@ public static Date getDate(Object o, PrimitiveObjectInspector oi) {
     case STRING:
       StringObjectInspector soi = (StringObjectInspector) oi;
       String s = soi.getPrimitiveJavaObject(o).trim();
-      try {
-        if (s.length() == DATE_LENGTH) {
-          result = Date.valueOf(s);
-        } else {
-          Timestamp ts = getTimestampFromString(s);
-          if (ts != null) {
-            result = Date.ofEpochMilli(ts.toEpochMilli());
-          }
+      if (s.length() == DATE_LENGTH) {
+        result = DateParser.parseDate(s);
+      } else {
+        Timestamp ts = getTimestampFromString(s);
+        if (ts != null) {
+          result = Date.ofEpochMilli(ts.toEpochMilli());
         }
-      } catch (IllegalArgumentException e) {
-        // Do nothing
```

Comment:
The net effect here for `Do nothing` is to return a `null` value.  `DateParser#parseDate` returns a `null` value instead of an Exception, so can remove this confusing try/catch block.